cnudie/util.py

```diff
@@ -589,3 +595,87 @@ def desanitise_version(version: str) -> str:
     desanitised_version = version.replace(META_SEPARATOR, '+')
 
     return desanitised_version
+
+
+def resource_content(
+    resource_node: 'cnudie.iter.ResourceNode',
+    oci_client: oci.client.Client,
```

Comment:
Rather make `oci_client` and `s3_client` optional attributes and only check their existence in case it is actually required for the resource's access type (or set reasonable defaults).

Reply:
in addition, I would prefer to pass-in an `access`, rather than a resource, or even a resource-node

even more specifically, I do not think this should be implemented in cc-utils. instead, have the bdba-function accept a acess-resolver-callback, and pass those in in using code -> delivery-service